docusaurus/docs/React/custom-code-examples/channel-users-list.mdx

```diff
@@ -0,0 +1,112 @@
+---
+id: channel-user-lists
+sidebar_position: 2
+title: Channel Members and Online Status
+---
+
+In this example, we will demonstrate how to render the current channel members and their online status.
+
+## Render the Channel Members List
+
+Let's start the example by creating a simple members list component. To access the members list of the current channel, we will get the current channel using `useChannelStateContext` hook. 
+The example is a bit more convoluted, since we will add online presence updates at the next step.
+
+:::note
+In order for the client to receive updates for user presence, ensure that you are watching the channel with `channel.watch({ presence: true })`. More details can be found in the [&nearr; LLC documentation](https://getstream.io/chat/docs/javascript/watch_channel/?language=javascript).
+:::
+
+```tsx
+const Users = () => {
+  const { channel } = useChannelStateContext();
+  const [channelUsers, setChannelUsers] = useState<
+    Array<{ name: string; online: boolean }>
+  >([]);
+
+  useEffect(() => {
+    const updateChannelUsers = (e?: Event) => {
+      // test if the updated user is a member of this channel
+      if (!e || channel.state.members[e.user!.id] !== undefined) {
+        setChannelUsers(
+          Object.values(channel.state.members).map((user) => ({
+            name: user.user_id!,
+            online: !!user.user!.online,
+          }))
+        );
+      }
+    };
+
+    updateChannelUsers();
+  }, [client, channel]);
+
+  return (
+    <ul>
+      {channelUsers.map((member) => (
+        <li key={member.name}>
+          {member.name} - {member.online ? "online" : "offline"}
+        </li>
+      ))}
+    </ul>
+  );
+};
+```
+
+We can put the component as a child of the `Channel` component:
+
+```tsx
+  <Channel>
+    <Window>
+      <Users />
+      <ChannelHeader />
+      <MessageList />
+      <MessageInput focus />
+    </Window>
+    <Thread />
+  </Channel>
+```
+
+## Real time updates 
+
+So far, our list looks good, but there's a catch: for performance purposes, the `useChannelStateContext` does not refresh when user presence changes. 
+To make the list refresh accordingly, we need to attach an additional listener to the `user.presence.changed` event of the chat client. 
```

Comment:
To the end of this sentence, you can also add. 'Let's also add some basic css to complete the look of the list.'

and then, this:

```css
.users-list {
  background: #ffffff;
  padding: 20px;
  padding-left: 30px;
  border-radius: calc(16px / 2) 16px 0 0;
  border: 1px solid #ecebeb;
}
```